src/reducers/appReducer.js

```diff
@@ -41,25 +42,25 @@ const makeAppReducer = () => {
 
         return;
 
-      case types.LOGIN_SUCCESS:
-        // TODO This is a Side effect.
-        // Is there a better place for setUserToStorage()?
+      case types.LOGIN_SUCCESS: {
         setUserToStorage(action.userData);
```

Comment:
I actually thought about this a couple of times, here's what my logic was:

Redux uses the event-driven pattern
To better understand this, think of events and events listeners in a web page.

When you click on something, a click event is dispatched in the browser.
If you want to listen to those events, you'll set up an event listener and you'll bind a callback to it.
So every time that event is dispatched, the callback you set up will be executed.
Now imagine you'd want to set up the same event callback for multiple event listeners.

It will probably look like this:
```javascript
function handleEvent(event) {
  switch (event.type) {
    case "mousemove":
      doSomething();
      break;

    case "click":
      doSomethingElse();
      break;
  }
}
```

Now what does that look like? (hint: redux reducer)

That being said, to do something when an event is dispatched, we'd do it in the callback, not in the dispatcher, right?

So as a parallel to that, redux does the same thing, we dispatch events, and the reducers act as event listeners. So it will only make sense to implement 3rd party functionality in the reducer, right?

Also, this is only valid because `setUserToStorage();` performs 3rd party functionality on the local storage, not on the redux store. If it was performing it on the redux store, then it would've been bad practice.

Hope this makes as much sense to you as it did to me 😬

Reply:
> So as a parallel to that, redux does the same thing, we dispatch events, and the reducers act as event listeners. So it will only make sense to implement 3rd party functionality in the reducer, right?

I don't agree with this. 
As I said we can do it as we are doing here in this specific case. Redux is unopinionated about this. It's just that we decided to keep reducers simple are pure. It's considered a best practice. There's a lot of literature about this.

We use thunks and action creators for this kind of side effects.

> Also, this is only valid because setUserToStorage(); performs 3rd party functionality on the local storage, not on the redux store. If it was performing it on the redux store, then it would've been bad practice.
 
🤔 Still a side effect. We just want to pass data to reducers and write a minimal amount of logic if it's needed. But we want them to be pure.

Happy to discuss it tomorrow if it doesn't make sense to you. This is just one approach, maybe the more "functional" one, but as I said, Redux gives plenty of freedom when it comes to decide these kind of things.

Reply:
By looking at how the pattern is supposed to be implemented, to me still makes more sense to do it in reducers.
But that's just my opionion, I'm willing to do whatever we decide on. The "recommended" redux way seems to be by using middleware, which kind of makes sense